build/scripts/Set-Options.ps1

```diff
@@ -0,0 +1,167 @@
+            Write-Warning "No options found for: $($command.path)"
+            continue
+        }
+
+        # Iterate around the matchend
+        foreach ($match in $result) {
+
+            # Ignore if this a lookup type
+            if ($type -ieq "lookup") {
+                continue
+            }
+
+            # get the subcommand name
+            $subcommand = $match.Groups["subcommand"].Value
+
+            # If the hashtable does not contain an entry for the command, add it
+            if (-not $rows.ContainsKey($subcommand)) {
+                $rows[$subcommand] = @()
+            }
+
+            # determine the the type of the input
+            $type = $match.Groups["type"].Value
+
+            # get the values of the definition
+            $data = ($match.Groups["definition"] -split ",") -replace "`"", ""
+
+            # Determine whether the input has a shortcut or not, this is done by
+            # checking the last character of the type
+            $short = ""
+            $field_name = 1
+            if ($type[-1] -eq 'P') {
+                $short = $data[2]
+                $field_default = 3
+                $field_description = 4
+            } else {
+                $field_default = 2
+                $field_description = 3
+            }
+
+            $name = $data[$field_name].Trim()
+            $default = $data[$field_default].Trim()
+            $description = [String]::IsNullOrEmpty($data[$field_description]) ? "" : $data[$field_description].Trim()
+
+            # using the row template set the row data
+            # define the array to be used for the substition in the template
+            $subs = @()
+
+            if ([String]::IsNullOrEmpty($short)) {
+                $subs += "``--{0}``" -f $name
+            } else {
+                $subs += "``--{0}``, ``-{1}``" -f $name, $short.Trim()
+            }
+
+            # determine if the paramneter is required, based on whether it has a default value or not
+            $required_indicator = "icon:times[fw]"
+            if ([String]::IsNullOrEmpty($default)) {
+                $required_indicator = "icon:check[fw]"
+            }
+
+            $subs += $required_indicator
+            $subs += $name.ToUpper()
+            $subs += $default
+            $subs += $description
+
+            $row = $template["row"] -f $subs
+
+            $rows[$subcommand] += $row
+
+        }
+
+        # Iterate around the keys in the rows to get the subcommands
+        foreach ($sub in $rows.Keys) {
+
+            # set the filename of the output file
+            $output_file = [IO.Path]::Combine($config.output, ("{0}.adoc" -f $sub))
+
+            # Create the table for this command
+            $table = $template["table"] -f ($rows[$sub] -join "`n")
+
+            # Finally write the table out to the file
+            Write-Information -MessageData "Writing: $($output_file)"
+            Set-Content -Path $output_file -Value $table
+        }
+
+    }
+
+# }
```

Comment:
Yar?

Reply:
I had this configured as a file that we could add to the EIR without modification, but that modification is slight so have removed the commented `function` declaration.